apulse: 0.1.10 -> 0.1.11, add pressureaudio

oxij opened the pull request:
###### Motivation for this change

Living without PulseAudio.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

oxij committed:
apulse: 0.1.10 -> 0.1.11

oxij committed:
libpressureaudio: init at `apulse.version`

oxij commented:
I tested `mpv` and `cmus`, both work fine with this and with plain ALSA disabled. `firefox` gets "permission denied" errors, probably because of sandboxing. More debugging needed there.
Sharing in case anyone else wants to play with this.

7c6f434c commented:
Firefox 58 is mentioned on the `apulse` homepage, there is an about:config option to loosen the sandbox and allow `ioctl`.

oxij commented:
Ah, thanks!

7c6f434c commented on pkgs/misc/apulse/pressureaudio.nix:
```diff
@@ -0,0 +1,81 @@
+  '';
+
+  installPhase = ''
+    echo "Coping libraries."
```
Either use `:` for empty phases and reduce verbosity here, or fix the typo: ‘‘Coping’’ → ‘‘Copying’’

7c6f434c commented on pkgs/misc/apulse/pressureaudio.nix:
```diff
@@ -0,0 +1,81 @@
+    echo "Coping libraries."
+    mkdir -p $out/lib
+    ls ${apulse}/lib/apulse $out/lib
+    cp -a ${apulse}/lib/apulse/* $out/lib/
```
Any reason for `cp -a` instead of `ln -s`?

7c6f434c commented on pkgs/misc/apulse/pressureaudio.nix:
```diff
@@ -0,0 +1,81 @@
+      But apulse is made to be used as a wrapper that substitutes its
+      replacement libs into LD_LIBRARY_PATH. The problem with that is
+      that you have to link against the original libpulseaudio, which
+      means you have to build it (and risk executing it eventually).
```
Maybe remove this paragraph (and the last one)? As it is, `libpulseaudio` packages in Nix does not build any executables, and I am not sure there are any risks in failed attempts to connect by PA client library.

oxij commented:
> Either use `:` for empty phases and reduce verbosity here, or fix the typo: ‘‘Coping’’ → ‘‘Copying’’

Oops, that and the correct .pc files were lost in the git stash. Fixed. Thanks.

> Any reason for `cp -a` instead of `ln -s`?

I tried, but it gets ugly fast as the original has symlinks you have to
preserve too. It's gonna be rehardlinked by nix-store store optimization
anyway. (Header files too, btw.)

> Maybe remove this paragraph (and the last one)? As it is, `libpulseaudio` packages in Nix does not build any executables, and I am not sure there are any risks in failed attempts to connect by PA client library.

Fixed.
